test/relativepath.spec.js

```diff
@@ -0,0 +1,39 @@
+const path = require('path')
+const chai = require('chai')
+const assert = chai.assert
+const loader = require('../index')
+
+describe('relativepath', function () {
+  [ 'yaml', 'json' ].forEach((type) => {
+    context(type, () => {
+      let thisScope;
+      beforeEach(function (done) {
+        //mock webpack loader this scope
+        const emptFn = function () {}
+        thisScope = {
+          addDependency: emptFn,
+          addContextDependency: emptFn,
+          cacheable: emptFn,
+          resource: path.join(__dirname, `./data/basic-app-context-${type}/locales/index.js`),
```

Comment:
Let's go ahead and rename this directory from `basic-app-context-` to the option name prefix `relativePathAsNamespace-`